### PR TITLE
Update e2e test to use the latest maven plugin

### PR DESCRIPTION
**Pull request by Flanker32**
Update e2etest to use the latest maven plugin , for the first part of issue #429

### CODE REVIEW

File: appveyor_function_e2e.yml [hunk 1]
```diff
@@ -58,3 +59,8 @@ test_script:
 on_finish:
   - ps: | 
       Stop-Process -Id $proc.Id -Erroraction Ignore
+      # Output standard input stream and error stream if fail
+      if (-not $success) {
```
Comment (pragnagopa) [inline appveyor_function_e2e.yml]: Useful to print these even if success

Comment (Flanker32) [reply appveyor_function_e2e.yml]: @pragnagopa done~

Comment (Flanker32) [reply appveyor_function_e2e.yml]: @pragnagopa fix done

File: e2etest.ps1 [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        Remove-Item -Recurse -Force $folderName
+    }
+}
+
+Function DownloadFileFromUrl($url, $destination) {
+    $wc = New-Object System.Net.WebClient
+    $wc.DownloadFile($Env:FUNCTIONCLI_URL, $functionCLIZipPath)
+}
+
+Function UpdateMavenPluginVersion($pomLocation, $version) {
```
Comment (SummerSun) [inline e2etest.ps1]: Why the second parameter never used?

File: e2etest.ps1 [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    $PomFile = gi $pomLocation
+    $Pom = [xml](gc $PomFile)
+    $Pom.project.properties.'azure.functions.maven.plugin.version' = $functionVersion
```
Comment (SummerSun) [inline e2etest.ps1]: mistake the $version with $functionVersion?

File: e2etest.ps1 [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    $Pom.Save($PomFile.Fullname)
+}
+
+#Scripts
+$base = pwd
+$functionCLIPath = "$base\Azure.Functions.Cli"
+$functionCLIZipPath = "$base\Azure.Functions.Cli.zip"
```
Comment (SummerSun) [inline e2etest.ps1]: `$functionCLIZipPath = $functionCLIPath +".zip"`. Once the function cli path changes, there is only one place to change.

**Comment by SummerSun** [comment]
LGTM now.

**Pull request merged**